cloudSupport/src/test/scala/cromwell/cloudsupport/aws/AwsConfigurationSpec.scala

```diff
@@ -0,0 +1,349 @@
+    customKey.name shouldBe "custom-keys"
+    customKey.accessKey shouldBe "access_key_id"
+    customKey.secretKey shouldBe "secret_key"
+
+    val assumeRoleWithId = (auths collectFirst { case a: AssumeRoleMode => a }).get
+    assumeRoleWithId.name shouldBe "assume-role-based-on-another-with-external"
+    assumeRoleWithId.baseAuthName shouldBe "default"
+    assumeRoleWithId.baseAuthentication.name shouldBe "default"
+    assumeRoleWithId.roleArn shouldBe "my-role-arn"
+    assumeRoleWithId.externalId shouldBe "my-external-id"
+
+    val assumeRole = (auths.takeRight(1) collectFirst { case a: AssumeRoleMode => a }).get
+    assumeRole.name shouldBe "assume-role-based-on-another"
+    assumeRole.baseAuthName shouldBe "default"
+    assumeRole.baseAuthentication.name shouldBe "default"
+    assumeRole.roleArn shouldBe "my-role-arn"
+    assumeRole.externalId shouldBe ""
+  }
+
+  it should "default region to us-east-1" in {
+    val config =
+      """|aws {
+         |  application-name = "cromwell"
+         |
+         |  auths = [
+         |    {
+         |      name = "name-default"
+         |      scheme = "default"
+         |    }
+         |  ]
+         |}
+         |""".stripMargin
+
+    val conf = AwsConfiguration(ConfigFactory.parseString(config))
+    conf.region shouldBe "us-east-1"
+  }
+
+  it should "return a known auth" in {
+    val config =
+      """|aws {
+         |  application-name = "cromwell"
+         |
+         |  auths = [
+         |    {
+         |      name = "name-default"
+         |      scheme = "default"
+         |    }
+         |  ]
+         |}
+         |""".stripMargin
+
+    val conf = AwsConfiguration(ConfigFactory.parseString(config))
+    conf.auth("name-default").map(_.name) should be("name-default".valid)
+  }
+
+  it should "not return an unknown auth" in {
+    val config =
+      """|aws {
+         |  application-name = "cromwell"
+         |
+         |  auths = [
+         |    {
+         |      name = "name-default"
+         |      scheme = "default"
+         |    }
+         |  ]
+         |}
+         |""".stripMargin
+
+    val conf = AwsConfiguration(ConfigFactory.parseString(config))
+    conf.auth("name-botched") should be(
+      "`aws` configuration stanza does not contain an auth named 'name-botched'.  Known auth names: name-default"
+        .invalidNel)
+  }
+
+  // it should "create an initializer with custom timeouts" in {
+  //   val transport = new MockHttpTransport()
+  //   val initializer = AwsConfiguration.withCustomTimeouts(request => {
+  //     request.getHeaders.set("custom_init", "ok")
+  //     ()
+  //   })
+  //   val factory = transport.createRequestFactory(initializer)
+  //   val request = factory.buildGetRequest(new GenericUrl(new URL("http://example.com")))
+  //   request.getConnectTimeout should be(180000)
+  //   request.getReadTimeout should be(180000)
+  //   request.getHeaders.get("custom_init") should be("ok")
+  // }
```

Comment:
?

Reply:
I kind of wanted to hold that test as a placeholder to use later, but it's good enough in history. I removed it in 4cc9600